File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/AvailabilityStatusMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.enums.HealthConnectAvailability
import com.rookmotion.rook_sdk_health_connect.data.proto.AvailabilityStatusProto

fun HealthConnectAvailability.toAvailabilityStatusProto(): AvailabilityStatusProto {
    return when (this) {
        HealthConnectAvailability.INSTALLED -> AvailabilityStatusProto.INSTALLED
        HealthConnectAvailability.NOT_INSTALLED -> AvailabilityStatusProto.NOT_INSTALLED
        HealthConnectAvailability.NOT_SUPPORTED -> AvailabilityStatusProto.NOT_SUPPORTED
    }
}

File: packages/rook_sdk_health_connect/lib/src/data/mapper/availability_status_mappers.dart
import 'package:rook_sdk_health_connect/src/data/proto/protos.pbenum.dart';
import 'package:rook_sdk_health_connect/src/domain/enums/hc_availability_status.dart';

extension AvailabilityStatusMappers on AvailabilityStatusProto {
  HCAvailabilityStatus toDomain() {
    return switch (this) {
      AvailabilityStatusProto.INSTALLED => HCAvailabilityStatus.installed,
      AvailabilityStatusProto.NOT_INSTALLED =>
        HCAvailabilityStatus.notInstalled,
      AvailabilityStatusProto.NOT_SUPPORTED =>
        HCAvailabilityStatus.notSupported,
      _ => throw Exception('Unknown availability status: $this'),
    };
  }
}
